src/ol/control/control.js

```diff
@@ -121,3 +124,18 @@ ol.control.Control.prototype.setMap = function(map) {
     map.render();
   }
 };
+
+
+/**
+ * This function is used to set a target element for the control. This
+ * function will have no effect if it is called after the control has
```

Comment:
It may also be important to maintain the control's position in the collection (meaning it might be unexpected to have it jump to the end of the collection when you call `setTarget`).

I imagine most people will expect `setTarget` to work when called more than once.  But I wouldn't be opposed to adding the method as is and then making it more useful later.